File: java/src/main/java/finance/clarian/sdk/CashOutService.java
package finance.clarian.sdk;

import java.util.LinkedHashMap;
import java.util.Map;

/** PIX cash-out: DICT lookup, create payout, fetch order. */
public final class CashOutService {

    private final ClarianClient client;

    CashOutService(ClarianClient client) {
        this.client = client;
    }

    /**
     * POST /pix/payouts/dict: preview a PIX key owner before paying out.
     *
     * @param keyType optional: CPF | CNPJ | EMAIL | PHONE | EVP
     */
    public Map<String, Object> dictCheck(String pixKey, String keyType) {
        Map<String, Object> body = new LinkedHashMap<>();
        body.put("pix_key", pixKey);
        if (keyType != null && !keyType.isEmpty()) {
            body.put("key_type", keyType);
        }
        return client.request("POST", "/pix/payouts/dict", body, null);
    }

    public Map<String, Object> dictCheck(String pixKey) {
        return dictCheck(pixKey, null);
    }

    /**
     * POST /cash-out/pix. {@code idempotencyKey} is required —
     * retries never double-send.
     */
    public Map<String, Object> create(Map<String, ?> params, String idempotencyKey) {
        return client.request("POST", "/cash-out/pix", params, idempotencyKey);
    }

    /** GET /cash-out/{id}. */
    public Map<String, Object> retrieve(String orderId) {
        return client.request("GET", "/cash-out/" + Paths.escape(orderId), null, null);
    }
}

File: dotnet/Clarian.Sdk/CashOutService.cs
namespace Clarian;

/// <summary>PIX cash-out: DICT lookup, create payout, fetch order.</summary>
public sealed class CashOutService
{
    private readonly ClarianClient _client;

    internal CashOutService(ClarianClient client) => _client = client;

    /// <summary>
    /// POST /pix/payouts/dict: preview a PIX key owner before paying out.
    /// keyType optional: CPF | CNPJ | EMAIL | PHONE | EVP
    /// </summary>
    public Task<DictCheckResponse> DictCheckAsync(
        string pixKey,
        string? keyType = null,
        CancellationToken cancellationToken = default)
    {
        var body = new DictCheckRequest
        {
            PixKey = pixKey,
            KeyType = string.IsNullOrEmpty(keyType) ? null : keyType,
        };
        return _client.RequestAsync<DictCheckResponse>(
            HttpMethod.Post,
            "/pix/payouts/dict",
            body,
            null,
            cancellationToken);
    }

    /// <summary>
    /// POST /cash-out/pix. idempotencyKey is required —
    /// retries never double-send.
    /// </summary>
    public Task<CashOutOrderResponse> CreateAsync(
        CashOutRequest request,
        string idempotencyKey,
        CancellationToken cancellationToken = default) =>
        _client.RequestAsync<CashOutOrderResponse>(
            HttpMethod.Post,
            "/cash-out/pix",
            request,
            idempotencyKey,
            cancellationToken);

    /// <summary>GET /cash-out/{id}.</summary>
    public Task<CashOutOrderResponse> RetrieveAsync(
        string orderId,
        CancellationToken cancellationToken = default) =>
        _client.RequestAsync<CashOutOrderResponse>(
            HttpMethod.Get,
            "/cash-out/" + ClarianClient.PathEscape(orderId),
            null,
            null,
            cancellationToken);
}
